tests/unittests/core/test_users_scraping.py

```diff
@@ -0,0 +1,387 @@
+"""Tests for the user scraping plugin system."""
+
+from collections.abc import Iterable
+from dataclasses import dataclass
+from importlib.metadata import EntryPoint, EntryPoints
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+
+from sarc.core.models.users import Credentials
+from sarc.core.scraping.users import (
+    MatchID,
+    UserMatch,
+    UserScraper,
+    _builtin_scrapers,
+    get_user_scraper,
+    scrape_users,
+    update_user_match,
+)
+
+
+class UserPluginTester:
+    plugin: UserScraper
+    raw_config: Any = None
+    parsed_config: Any = {}
+
+    def test_config_validation(self):
+        conf = self.plugin.validate_config(self.raw_config)
+        assert isinstance(conf, self.plugin.config_type)
+        assert conf == self.parsed_config
+
+    def test_fetch_data(self):
+        raise NotImplementedError()
+
+    def test_parse_data(self, raw_data, data_regression):
+        raise NotImplementedError()
+        # This code is what you should put in your implementation, but
+        # limitations on parametrize require that the method be in your class
+        # rather than this one.
+        data = self.plugin.parse_user_data(self.parsed_config, raw_data)
+        data_regression.check(data)
+
+
+@dataclass
+class MockConfig:
+    api_url: str
+    api_key: str
+
+
+class MockUserScraper(UserScraper[MockConfig]):
+    config_type = MockConfig
+
+    def get_user_data(self, config: MockConfig) -> bytes:
+        return f"mock_data_from_{config.api_url}".encode()
+
+    def parse_user_data(self, config: MockConfig, data: bytes) -> Iterable[UserMatch]:
+        users = []
+
+        user1 = UserMatch(
+            display_name="John Doe",
+            email="john.doe@example.com",
+            matching_id=MatchID(name="mock_plugin", mid="user1"),
+        )
+        user1.github_username.insert("johndoe")
+        user1.google_scholar_profile.insert("scholar.google.com/citations?user=abc123")
+        users.append(user1)
+
+        user2 = UserMatch(
+            display_name="Jane Smith",
+            email=None,
+            matching_id=MatchID(name="mock_plugin", mid="user2"),
+        )
+        users.append(user2)
+
+        user3 = UserMatch(
+            display_name="Bob Wilson",
+            email="bob.wilson@example.com",
+            matching_id=MatchID(name="mock_plugin", mid="user3"),
+            known_matches={
+                MatchID(name="other_plugin", mid="bob_wilson"),
+                MatchID(name="another_plugin", mid="bwilson"),
+            },
+        )
+        users.append(user3)
+
+        return users
+
+
+test_entry_points = EntryPoints(
+    [
+        EntryPoint(
+            name="test_plugin",
+            group="sarc.user_scraper",
+            value=f"{MockUserScraper.__module__}:{MockUserScraper.__name__}",
+        )
+    ]
+)
+
+
+@pytest.fixture
+def user_plugin(monkeypatch):
```

Comment:
We may need a fixture `user_plugins` with at least 2 scrapers, and some associated unit tests to check if scraping works correctly when multiple scrapers are available.